frontend/packages/dev-console/src/components/progressive-list/ProgressiveList.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import FormSectionDivider from '../import/section/FormSectionDivider';
+import ProgressiveListFooter from './ProgressiveListFooter';
+import './ProgressiveList.scss';
+import ProgressiveListItem from './ProgressiveListItem';
+
+export interface ProgressiveListProps {
+  visibleItems: string[];
+  onVisibleItemChange: (item: string) => void;
+  text: string;
+}
+
+const ProgressiveList: React.FC<ProgressiveListProps> = ({
+  visibleItems,
+  children,
+  onVisibleItemChange,
+  text,
+}) => {
+  const items: string[] = [];
+  return (
+    <React.Fragment>
+      {React.Children.map(children, (child: React.ReactElement, index: number) => {
+        if (child.type !== ProgressiveListItem) return null;
+        const { name } = child.props;
+        const isNameInVisibleItems = visibleItems.includes(name);
+        if (!isNameInVisibleItems) {
+          items.push(name);
+        }
+        return isNameInVisibleItems ? (
+          <React.Fragment>
+            {child}
+            {index !== React.Children.count(children) - 1 && <FormSectionDivider />}
```

Comment:
hmm this won't always be true if the last react elements aren't proper `PropressiveListItem`.
You might need to first filter the list of children to that of only the proper types and then do the map.
Or do the map then filter the nulls and then map over that list in the render return value.

Reply:
fixed in https://github.com/openshift/console/pull/1772/commits/243479b9e72d610bd4e122486e5208a2a256d6ba

Reply:
fixed in https://github.com/openshift/console/pull/1772/commits/180fa305e23b013d1e83a0051f678f9201ef22f5